clarifai/datasets/upload/utils.py

```diff
@@ -41,3 +45,200 @@ def load_module_dataloader(module_dir: Union[str, os.PathLike], **kwargs) -> Cla
+
+
+class DisplayUploadStatus:
+  """Class to display dataset upload status."""
+
+  def __init__(self, dataloader: ClarifaiDataLoader,
+               dataset_metrics_response: Type[MultiDatasetVersionMetricsGroupResponse],
+               dataset_info_dict: Dict[str, str]) -> None:
+    """Initialize the class.
+    Args:
+      dataloader: ClarifaiDataLoader object
+      dataset_metrics_response: The dataset version metrics response from the server.
+      dataset_info_dict: The dataset info dictionary.
+    """
+    self.dataloader = dataloader
+    self.dataset_metrics_response = dataset_metrics_response
+    self.dataset_info_dict = dataset_info_dict
+
+    self.display()
+
+  def display(self) -> None:
+    """Display the upload status."""
+    from rich.console import Console
+
+    local_inputs_count, local_annotations_dict = self.get_dataloader_stats()
+    uploaded_inputs_dict, uploaded_annotations_dict = self.get_uploaded_dataset_stats()
+
+    self.local_annotations_count = sum(local_annotations_dict.values())
+    self.uploaded_annotations_count = sum(uploaded_annotations_dict.values())
+
+    local_dataset_dict = {
+        "Inputs Count": str(local_inputs_count),
+        "Annotations Count": str(local_annotations_dict)
+    }
+    uploaded_dataset_dict = {
+        "Inputs Count": str(uploaded_inputs_dict["total"]),
+        "Annotations Count": str(uploaded_annotations_dict)
+    }
+
+    panel_layout = self.get_display_layout(local_dataset_dict, uploaded_dataset_dict)
+
+    console = Console()
+    console.print(panel_layout)
+
+  def get_dataloader_stats(self) -> Tuple[int, Dict[str, int]]:
+    """Get the number of inputs and annotations in a dataloader.
+
+    Returns:
+      local_inputs_count (int): total number of inputs in the dataloader
+      local_annotations_dict (Dict[str, int]): total number of annotations in the dataloader
+    """
+    from clarifai.constants.dataset import DATASET_UPLOAD_TASKS
+
+    if not isinstance(self.dataloader, ClarifaiDataLoader):
+      raise UserError("Dataloader is not an instance of ClarifaiDataLoader")
+
+    task = self.dataloader.task
+    if task not in DATASET_UPLOAD_TASKS:
+      raise UserError(
+          "Invalid task, please use one of the following: {}".format(DATASET_UPLOAD_TASKS))
+    local_inputs_count = len(self.dataloader)
+    local_annotations_dict = dict(concepts=0, bboxes=0, polygons=0)
+    for i in range(local_inputs_count):
+      if task == "visual_segmentation":
+        local_annotations_dict["polygons"] += len(self.dataloader[i].polygons)
+      elif task == "visual_detection":
+        local_annotations_dict["bboxes"] += len(self.dataloader[i].bboxes)
+      elif task == "text_classification":
+        local_annotations_dict["concepts"] += len(self.dataloader[i].labels)
+      elif task == "visual_classification" or task == "visual_captioning":
+        local_annotations_dict["concepts"] += len(self.dataloader[i].labels)
```

Comment:
as more tasks and annotation types may be supported in the future, we could define a dict that maps the task -> annot_dict key -> attribute from data_item. We can then refactor this to simply read from the dict and use getattr on the data_item. e.g.
```python
## Put this in constants file somewhere
some_dict = {
    "visual_segmentation": {"polygons": "polygons"},
    "some_task": {"key": "attr"}
}
.....

## This for loop
for ...... :
    key, attr = [(k, v) for k, v in some_dict.get(task).items()][0]
    local_annotations_dict[key] += len(getattr(self.dataloader[i], attr))
```

Something like that. That way we will not have to add an `elif` every time we get a new annotation type or a task type.

Reply:
Added this